pallets/metaverse/src/lib.rs

```diff
@@ -736,8 +739,45 @@ impl<T: Config> Pallet<T> {
 			};
 			Some(v2)
 		});
+		log::info!("{} metaverses upgraded:", upgraded_metaverse_items);
+		0
+	}
 
-		log::info!("{} metaverses upgraded:", num_metaverse_items);
+	pub fn upgrade_metaverse_info_v3() -> Weight {
+		log::info!("Start upgrade_metaverse_info_v3");
+		let mut upgraded_metaverse_items = 0;
+		let mut total_metaverse_items = 0;
+
+		Metaverses::<T>::translate(|k, metaverse_info_v2: MetaverseInfo<T::AccountId>| {
+			total_metaverse_items += 1;
+			if metaverse_info_v2.land_class_id == 0 && metaverse_info_v2.estate_class_id == 1 {
+				let new_land_class_id = Self::mint_metaverse_land_class(&metaverse_info_v2.owner, k).unwrap_or({
+					log::info!("Cannot create land class for metaverse {}:", total_metaverse_items);
+					return None;
+				});
+				let new_estate_class_id = Self::mint_metaverse_estate_class(&metaverse_info_v2.owner, k).unwrap_or({
+					log::info!("Cannot create estate class for metaverse {}:", total_metaverse_items);
+					return None;
+				});
+
+				upgraded_metaverse_items += 1;
+
+				let v3: MetaverseInfo<T::AccountId> = MetaverseInfo {
+					owner: metaverse_info_v2.owner,
+					metadata: metaverse_info_v2.metadata,
+					currency_id: metaverse_info_v2.currency_id,
+					is_frozen: false,
+					listing_fee: Perbill::from_percent(0u32),
+					land_class_id: new_land_class_id,
+					estate_class_id: new_estate_class_id,
+				};
+				Some(v3)
+			} else {
+				return None;
```

Comment:
Return None will delete the actual record of metaverse info. It should return Some(metaverse_info_v2) if it's none change.

Reply:
Thanks for pointing this out. I have just updated the code